build/gulp/tasks/stats.ts

```diff
@@ -0,0 +1,100 @@
+import * as fs from 'fs'
+import * as glob from 'glob'
+import { src, task, parallel, series } from 'gulp'
+import * as gzip from 'gzip-size'
+import * as _ from 'lodash'
+import * as prettyBytes from 'pretty-bytes'
+import * as uglifyES from 'uglify-es'
+
+import config from '../../../config'
+
+const { version } = require('../../../package.json')
+const { paths } = config
+
+// ----------------------------------------
+// Stats
+// ----------------------------------------
+
+task('build:stats', cb => {
```

Comment:
We loop over each component info file in `docs/src/componentInfo`.  We keep totals of components, subcomponents, and their max, min, and gzip sizes as we loop through them.  These are keyed by package.json version number.